core/modules/query/QueryContext.cc

```diff
@@ -52,35 +155,65 @@ QueryContext::resolve(std::shared_ptr<ColumnRef> cr) {
             if (concrete.db.empty()) {
                 concrete.db = defaultDb;
             }
-            return concrete;
+            dbTableSet.insert(concrete);
+            return dbTableSet;
         }
     }
     // Set default db and table.
     DbTablePair p;
-    if (cr->table.empty()) { // No db or table: choose first resolver pair
-        p = resolverTables[0];
-        // TODO: We can be fancy and check the column name against the
-        // schema for the entries on the resolverTables, and choose
-        // the matching entry.
+    if (cr->table.empty()) { // No db or table.
+        if (columnToTablesMap.empty()) {
+            // This should only be the case if no connection to the database could be made, which
+            // should only happen in unit tests. This should get through the unit tests and have
```

Comment:
"which should only happen in unit tests"

Or if connection to database fails for any other reason?